scripts/main.js

```diff
@@ -67,8 +65,16 @@
 			}
 		}
 
-		xmpp_uri_parts[0] = encodeURIComponent(xmpp_uri_parts[0]) // URL-encode the JID only
-		var xmpp_uri_encoded = xmpp_uri_parts.join("?");
+		var jid_parts = xmpp_uri_parts[0].split("@");
```

Comment:
You might want to consistently use `const` instead of `var`. In the (hopefully rare) situations where you want to allow reassignment, use `let`.

The scoping of `var` variables is not intuitive which can lead to bugs that are difficult to debug